ocaml/testsuite/tests/typing-implicit-source-positions/fn_decl_and_defn.ml

```diff
@@ -0,0 +1,67 @@
+(* TEST
+   * expect
+*)
+
+type t = src_pos:[%src_pos] -> unit -> unit
+
+[%%expect {|
+type t = src_pos:lexing_position -> unit -> unit
```

Comment:
That makes sense, I think I printed the wrong thing here. Fixed in next commit

Reply:
That makes sense, the erroneous occurrences of `lexing_position` for function definitions were because I printed the wrong type, and the underlying type is okay. I'll fix this portion next. I think it is because I forgot to amend Tarrow after adding Position, as mentioned on slack